service/history/workflow/update/registry.go

```diff
@@ -219,6 +227,51 @@ func (r *registry) FindOrCreate(ctx context.Context, id string) (*Update, bool,
 	return upd, false, nil
 }
 
+func (r *registry) TryResurrect(ctx context.Context, acptOrRejMsg *protocolpb.Message) (*Update, error) {
+	if acptOrRejMsg == nil || acptOrRejMsg.Body == nil {
+		return nil, nil
+	}
+
+	// Check only total limit here. This might add more than maxInFlight updates to registry,
+	// but provides better developer experience.
+	if err := r.checkTotalLimit(ctx); err != nil {
+		return nil, err
+	}
+
+	body, err := acptOrRejMsg.Body.UnmarshalNew()
+	if err != nil {
+		return nil, invalidArgf("unable to unmarshal request: %v", err)
+	}
+	err = utf8validator.Validate(body, utf8validator.SourceRPCRequest)
+	if err != nil {
+		return nil, invalidArgf("unable to unmarshal request: %v", err)
```

Comment:
Should this error message say something about utf8 validation?

Reply:
It will say, because `err` will have it, but I agree it is better to be more specific here. I fixed it here and other place where I copied this from.